cmd/bcpd/app/app_test.go

```diff
@@ -296,36 +226,102 @@ func TestApp(t *testing.T) {
 	assert.EqualValues(t, 2, c.ActivationThreshold)
 	assert.EqualValues(t, 2, c.AdminThreshold)
 
+	// create a wallet at contractAddr
+	msg = &cash.SendMsg{
+		Src:  addr,
+		Dest: contractAddr,
+		Amount: &x.Coin{
+			Whole:  2000,
+			Ticker: "ETH",
+		},
+		Memo: "New wallet controlled by a contract",
+	}
+	tx = &Tx{
+		Sum: &Tx_SendMsg{msg},
+	}
+	_, cres = signAndCommit(t, myApp, tx, []Signer{{pk, 2}}, chainID, 4)
+	assert.NotEmpty(t, cres.Data)
+
 	// build and sign a transaction with contract
 	msg = &cash.SendMsg{
 		Src:  contractAddr,
 		Dest: addr2,
 		Amount: &x.Coin{
-			Whole:  2000,
+			Whole:  1000,
 			Ticker: "ETH",
 		},
-		Memo: "Have a great trip!",
+		Memo: "Gift from a contract!",
 	}
 	tx = &Tx{
 		Sum:      &Tx_SendMsg{msg},
 		Multisig: contractID,
 	}
-	sig1, err = sigs.SignTx(pk, tx, chainID, 1)
-	require.NoError(t, err)
-	sig2, err := sigs.SignTx(pk2, tx, chainID, 0)
-	require.NoError(t, err)
-	tx.Signatures = []*sigs.StdSignature{sig1, sig2}
-	txBytes, err = tx.Marshal()
+	_, cres = signAndCommit(t, myApp, tx, []Signer{{ck1, 0}, {ck2, 0}}, chainID, 5)
+	assert.NotEmpty(t, cres.Data)
+
+	// make sure money arrived safely
+	contractAddrKey := namecoin.NewWalletBucket().DBKey(contractAddr)
+	cwquery := abci.RequestQuery{
+		Path: "/",
+		Data: contractAddrKey,
+	}
+	cwqres := myApp.Query(cwquery)
+	checkWalletQuery(t, cwqres, "", 1, map[int]int64{0: 1000}, map[int]string{0: "ETH"})
+}
+
+type Signer struct {
+	pk    *crypto.PrivateKey
+	nonce int64
+}
+
+// signAndCommit signs tx with signatures from signers and submits to the chain
+// asserts and fails the test in case of errors during the process
+func signAndCommit(t *testing.T, app app.BaseApp, tx *Tx, signers []Signer, chainID string, blockHeight int64) (abci.ResponseDeliverTx, abci.ResponseCommit) {
```

Comment:
Looks good, but I don't think we need ResponseCommit. Or do you use that for the height?

Reply:
It just seems to be used for `assert.NotEmpty(t, cres.Data)`.
You can add that test in here, and then remove that from return values